tracer/src/Datadog.Trace/Configuration/ImmutableGCPFunctionSettings.cs

```diff
@@ -0,0 +1,25 @@
+// <copyright file="ImmutableGCPFunctionSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.Configuration
+{
+    /// <summary>
+    /// Settings class for gathering metadata about the execution context in Google Cloud Functions
+    /// References:
+    /// https://cloud.google.com/functions/docs/configuring/env-var#runtime_environment_variables_set_automatically
+    /// </summary>
+    internal class ImmutableGCPFunctionSettings
```

Comment:
I don't think we need a full-blown `Immutable*Settings` class for a helper static method. These types are meant to hold non-static configuration data. `GetIsGCPFunction()` and `GetIsAzureConsumptionPlanFunction()` could probably live in a static class `/ClrProfiler/ServerlessInstrumentation`.

But not really a blocker for this PR. Maybe something to cleanup later.

At least let's make this class `static` to prevent unnecessary instantiation.

```suggestion
    internal static class ImmutableGCPFunctionSettings
```